HeterogeneousCore/Producer/test/TestHeterogeneousEDProducerGPUHelpers.cu

```diff
@@ -1,11 +1,11 @@
-#include "TestHeterogeneousEDProducerGPUHelpers.h"
-
-#include "FWCore/MessageLogger/interface/MessageLogger.h"
-
-#include "cuda/api_wrappers.h"
+#include <cuda/api_wrappers.h>
 #include <cuda.h>
 #include <cuda_runtime.h>
 
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+#include "TestHeterogeneousEDProducerGPUHelpers.h"
```

Comment:
Let me note once that there is a technical argument for including the interface header as the first one in the implementation file. It checks that the header itself includes all other headers needed to compile it.

Reply:
I have heard about it, but it feels "untidy" to `#include` files in inverse order of generality.

From the technical point of view, I actually prefer to `#include` the more general header first, to minimise the chance that they `#define` something that alters the behaviour of ones included after.

Anyway, once CMSSW has an agreed-upon policy (via clang-tidy and/or clang-format) I'll abide to it... for the moment I prefer to include things from the most general.